src/lava/lib/optimization/utils/state_analyzer.py

```diff
@@ -0,0 +1,225 @@
+# Copyright (C) 2022 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import typing as ty
+import warnings
+
+import numpy as np
+import seaborn as sns
+from lava.lib.optimization.problems.problems import QUBO
+from lava.lib.optimization.solvers.generic.solver import SolverReport
+from matplotlib import pyplot as plt
+from matplotlib.figure import Figure
+
+sns.set_style("whitegrid")
+
+
+class StateAnalyzer:
+    """
+    Utility class to analyze and plots useful metrics on the execution of
```

Comment:
plot instead of plots

Reply:
Maybe also phrase it in a way that explains the presence of "problem" as input parameter  in the __init__ method